pkg/grpc/podstore/podstore.go

```diff
@@ -246,7 +246,7 @@ func (s store) MarkPodFailed(_ context.Context, req *podstore_protos.MarkPodFail
 	// we don't really need the CAS properties of MutateStatus but there
 	// should be only one system trying to write the status record so it
 	// doesn't hurt.
-	err = s.podStatusStore.MutateStatus(podUniqueKey, mutator)
+	err = s.podStatusStore.MutateStatus(ctx, podUniqueKey, mutator)
```

Comment:
it's safe to cancel a context multiple times. and yeah i think all `transaction.New()` should be followed with that `defer`.

I can't remember what doc i was looking at that said you can cancel multiple times, but it's clear from the source code:

```
func (c *cancelCtx) cancel(removeFromParent bool, err error) {
  	if err == nil {
  		panic("context: internal error: missing cancel error")
  	}
  	c.mu.Lock()
  	if c.err != nil {
  		c.mu.Unlock()
  		return // already canceled
  	}
...
```
(https://golang.org/src/context/context.go?s=8329:8393#L335)